src/charm.py

```diff
@@ -433,7 +332,7 @@ def _configure(self, _: EventBase) -> None:  # noqa: C901
         if not (vault := self._get_active_vault_client()):
             return
         self._configure_pki_secrets_engine()
-        self._sync_vault_autounseal()
+        self._sync_vault_autounseal(vault)
```

Comment:
Why does `_sync_vault_autounseal` take the vault client as an argument and the other syncs (ex. kv) get their vault client inside on their own?